### PR TITLE
[bitnami/*] Improve logic for retries workflow

### PR DESCRIPTION
**Pull request by FraPazGal**
### Description of the change

Fixes a typo on the POST API URL to avoid current `404` [errors](https://github.com/bitnami/charts/actions/runs/6482224207/job/17601314275). 

The PR now includes some additional improvements:

- Removed an unnecessary usage of the GitHub token for a public API call. 
- Parametrised the `$MAX_RUNS_ATTEMPS` and `$MAX_RETRY_SLOTS` variables.
- The "List workflow runs" query is now limited to runs launched by the `bitnami-bot` in the last 3 days.
- Increased the number of results from the "List workflow runs" query to the maximum of `100` per page.
- Increased workflow's verbose.

These changes we'll allow us to avoid retrieving unnecessary or obsolete info and will make it harder to reach the pagination limit of 100 results.

This leaves untreated the corner case of a catalog-wide error in VIB testing during a catalog-wide release. Given the low probability of this scenario, for now we'll avoid adding more complexity to the current logic by implementing pagination management.

### Benefits

N/A

### Possible drawbacks

N/A

### Additional information

Changes tested at: https://github.com/FraPazGal/charts/actions/runs/6534809694/job/17742853360

### Checklist

- [x] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: .github/workflows/retry-failed-releases.yml [hunk 1]
```diff
@@ -21,10 +21,10 @@ jobs:
...
         run: |
           # Obtain "CI Pipeline" failed runs and filter those from release PRs with less than 3 attempts
-          curl -H 'Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}' -X GET -Lkso ${{ env.TEMP_FILE }} ${{ env.ACTIONS_API_ENDPOINT }}/workflows/${{ env.WORKFLOW_ID }}/runs?status=failure
+          curl -X GET -Lkso ${{ env.TEMP_FILE }} ${{ env.ACTIONS_API_ENDPOINT }}/workflows/${{ env.WORKFLOW_ID }}/runs?status=failure
           readarray -t failed_runs_ids < <(jq '.workflow_runs[] | select((.run_attempt < 3) and (.display_title | contains("Release")) and (.head_commit.author.email=="bitnami-bot@vmware.com")).id' ${{ env.TEMP_FILE }})
 
           for run_id in "${failed_runs_ids[@]:0:15}"; do
             echo "Retrying workflow $(jq --argjson id $run_id '.workflow_runs[] | select(.id==$id) | .html_url' ${{ env.TEMP_FILE }})"
-            curl -H 'Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}' -X POST -Lks ${{ env.ACTIONS_API_ENDPOINT }}/workflows/runs/${run_id}/rerun
+            curl -H 'Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}' -X POST -Lks ${{ env.ACTIONS_API_ENDPOINT }}/runs/${run_id}/rerun
```
Comment (fmulero) [inline .github/workflows/retry-failed-releases.yml:29]: NOTE: We could take directly the `rerun_url` field from the ["list workflow runs" response](https://docs.github.com/en/rest/actions/workflow-runs?apiVersion=2022-11-28#list-workflow-runs-for-a-workflow)

**Review by fmulero — APPROVED** [review]
Thanks a lot @FraPazGal! LGTM

I left a little comment just for information.

**Review by Mauraza — DISMISSED** [review]

File: .github/workflows/retry-failed-releases.yml (hunk 1 above)
Comment (FraPazGal) [reply .github/workflows/retry-failed-releases.yml]: Good idea! I have updated the PR and also added some miscellaneous improvements. More info on the updated task description.

**Review by joancafom — APPROVED** [review]

**Review by fmulero — APPROVED** [review]
LGTM

**Pull request merged**